client/src/components/ApplicationList.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+import { Job, Application } from '../lib';
+import JobApplicationCard from './JobApplicationCard';
+import Pagination from './Pagination';
+
+type ApplicationListProps = {
+  jobApplication: (Application & Job)[] | undefined;
```

Comment:
Do you use `Application & Job` more than this once? If so, you might want to declare it as a type (say, JobApplication).